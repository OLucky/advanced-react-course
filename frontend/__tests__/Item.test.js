import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';

import ItemComponent from '../components/Item';

const fakeItem = {
  id: '11111',
  title: 'Cool thing',
  price: 4000,
  description: 'Some test',
  image: 'stuff.jpg',
  largeImage: 'stuff-large.jpg'
}

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem}/>);

    expect(toJSON(wrapper)).toMatchSnapshot();
  })

  // it('renders price tag properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem}/>);
  //   const PriceTag = wrapper.find('PriceTag');
    
  //   expect(PriceTag.children().text()).toBe('$50');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });

  // it('renders the image', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem}/>);
  //   const img = wrapper.find('img');

  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });

  // it('renders out the buttons', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem}/>);
  //   const ButtonList = wrapper.find('.buttonList');

  //   expect(ButtonList.children()).toHaveLength(3);
  //   expect(ButtonList.find('Link')).toHaveLength(1);
  //   expect(ButtonList.find('DeleteItem')).toHaveLength(1);
  //   expect(ButtonList.find('AddToCart')).toHaveLength(1);
  // })
});