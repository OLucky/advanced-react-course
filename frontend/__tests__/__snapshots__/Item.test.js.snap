// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Cool thing"
    src="stuff.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "11111",
          },
        }
      }
    >
      <a>
        Cool thing
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    Some test
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "11111",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="11111"
    />
    <DeleteItem
      id="11111"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
